src/release-specification.ts

```diff
@@ -301,6 +299,58 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release new versions of dependencies alongside their dependents
+      if (
+        pkg &&
+        versionSpecifierOrDirective &&
+        (hasProperty(IncrementableVersionParts, versionSpecifierOrDirective) ||
+          isValidSemver(versionSpecifierOrDirective))
+      ) {
+        const missingDependencies = Object.keys({
+          ...(pkg.unvalidatedManifest.dependencies || {}),
+          ...(pkg.unvalidatedManifest.peerDependencies || {}),
+        }).filter((dependency) => {
+          const dependencyVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[dependency];
+
+          return (
+            dependencyVersionSpecifierOrDirective !== SKIP_PACKAGE_DIRECTIVE &&
+            dependencyVersionSpecifierOrDirective !==
+              INTENTIONALLY_SKIP_PACKAGE_DIRECTIVE &&
+            !hasProperty(
+              IncrementableVersionParts,
+              dependencyVersionSpecifierOrDirective,
+            ) &&
+            !isValidSemver(dependencyVersionSpecifierOrDirective)
+          );
```

Comment:
Following Mark's comment, I'm wondering whether we can make this conditional simpler than it is, knowing that:

- if the release spec is missing a package, then attempting to look it up should return `undefined`.
- if a version was never specified for a package and it's not being intentionally skipped, then it should be `null`.

Therefore we should be able to say:

``` suggestion
          return (
            dependencyVersionSpecifierOrDirective === undefined ||
            dependencyVersionSpecifierOrDirective === SKIP_PACKAGE_DIRECTIVE
          );
```

Reply:
actually the error should be thrown in case dependencyVersionSpecifierOrDirective === undefined or dependencyVersionSpecifierOrDirective === null I updated the PR to !dependencyVersionSpecifierOrDirective in this case